packages/react/src/common/is-equal.ts

```diff
@@ -0,0 +1,14 @@
+/**
+ * Deeply compare two values to determine if they are equal.
+ * Supports primitives and serializable objects.
+ * @param {unknown} value First value to compare
+ * @param {unknown} other Second value to compare
+ * @returns {boolean} True if the values are equal
+ */
+export function isEqual(value: unknown, other: unknown): boolean {
+  if (value === other) {
+    return true;
+  }
+
+  return JSON.stringify(value) === JSON.stringify(other);
```

Comment:
For performance, I don't think we should attempt this serialization if we're dealing with a primitive. Currently, this will attempt things like  `JSON.stringify(1) === JSON.stringify(2)`;

Reply:
I've refactored the check.